core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/PythonWorkflowWorker.scala

```diff
@@ -23,26 +29,48 @@ import scala.sys.process.{BasicIO, Process}
 object PythonWorkflowWorker {
   def props(
       id: ActorVirtualIdentity,
-      op: IOperatorExecutor,
+      workerIdx: Int,
+      opFn: (Int) => IOperatorExecutor,
       parentNetworkCommunicationActorRef: NetworkSenderActorRef,
       allUpstreamLinkIds: Set[LinkIdentity]
   ): Props =
-    Props(new PythonWorkflowWorker(id, op, parentNetworkCommunicationActorRef, allUpstreamLinkIds))
+    Props(
+      new PythonWorkflowWorker(
+        id,
+        workerIdx,
+        opFn,
+        parentNetworkCommunicationActorRef,
+        allUpstreamLinkIds
+      )
+    )
 }
 
 class PythonWorkflowWorker(
     actorId: ActorVirtualIdentity,
-    operator: IOperatorExecutor,
+    workerIdx: Int,
+    opFn: (Int) => IOperatorExecutor,
     parentNetworkCommunicationActorRef: NetworkSenderActorRef,
     allUpstreamLinkIds: Set[LinkIdentity]
 ) extends WorkflowWorker(
       actorId,
-      operator,
+      workerIdx,
+      opFn,
       parentNetworkCommunicationActorRef,
       allUpstreamLinkIds,
       false
     ) {
-
+  pythonProxyClient.enqueueCommand(
+    ControlInvocation(
+      IgnoreReply,
+      InitializeOperatorLogic(
+        operatorExecutor.asInstanceOf[PythonUDFOpExecV2].getCode,
+        allUpstreamLinkIds.toArray,
+        operatorExecutor.isInstanceOf[ISourceOperatorExecutor],
+        operatorExecutor.asInstanceOf[PythonUDFOpExecV2].getOutputSchema
+      )
+    ),
+    SELF
+  )
```

Comment:
I think it is a minor concern because this message will stay in the queue if the proxy is not initialized yet. But I can move it to `preStart` call if you want.

Reply:
I think it may be better to move it after the proxy layer gets initialized. I am not sure if `preStart` is the correct location either.